pkg/oc/cli/cmd/export.go

```diff
@@ -62,64 +92,79 @@ func NewCmdExport(fullName string, f *clientcmd.Factory, streams genericclioptio
 		Deprecated: "use the oc get --export",
 		Hidden:     true,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunExport(f, exporter, streams.In, streams.Out, cmd, args, filenames)
-			if err == kcmdutil.ErrExit {
-				os.Exit(1)
-			}
-			kcmdutil.CheckErr(err)
+			kcmdutil.CheckErr(o.Complete(f, args))
+			kcmdutil.CheckErr(o.Validate(cmd))
+			kcmdutil.CheckErr(o.Run())
 		},
 	}
 
-	cmd.Flags().String("as-template", "", "Output a Template object with specified name instead of a List or single object.")
-	cmd.Flags().Bool("exact", false, "If true, preserve fields that may be cluster specific, such as service clusterIPs or generated names")
-	cmd.Flags().Bool("raw", false, "If true, do not alter the resources in any way after they are loaded.")
-	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on")
-	cmd.Flags().Bool("all-namespaces", false, "If true, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
-	cmd.Flags().StringSliceVarP(&filenames, "filename", "f", filenames, "Filename, directory, or URL to file for the resource to export.")
+	o.PrintFlags.AddFlags(cmd)
+
+	cmd.Flags().StringVar(&o.AsTemplateName, "as-template", o.AsTemplateName, "Output a Template object with specified name instead of a List or single object.")
+	cmd.Flags().BoolVar(&o.Exact, "exact", o.Exact, "If true, preserve fields that may be cluster specific, such as service clusterIPs or generated names")
+	cmd.Flags().BoolVar(&o.Raw, "raw", o.Raw, "If true, do not alter the resources in any way after they are loaded.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "If true, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().StringSliceVarP(&o.Filenames, "filename", "f", o.Filenames, "Filename, directory, or URL to file for the resource to export.")
 	cmd.MarkFlagFilename("filename")
 	cmd.Flags().Bool("all", true, "DEPRECATED: all is ignored, specifying a resource without a name selects all the instances of that resource")
 	cmd.Flags().MarkDeprecated("all", "all is ignored because specifying a resource without a name selects all the instances of that resource")
-	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVar(&o.OutputVersion, "output-version", o.OutputVersion, "The preferred API versions of the output objects")
```

Comment:
@soltysh ideally this whole command would be deprecated in the future - so I don't think we should worry about not supporting output-version specifically

Reply:
Yeah, but that'd also mean we're breaking backwards compatibility and I don't think we want to do that.